daffodil-io/src/main/scala/org/apache/daffodil/processors/charset/IBM1047.scala

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.daffodil.processors.charset
+
+import org.apache.daffodil.io.InputSourceDataInputStream
+import org.apache.daffodil.io.FormatInfo
+
+object BitsCharsetIBM1047 extends {
+  override val name = "IBM1047"
+  override val aliases = Seq("X-IBM-1047-S390")
```

Comment:
Why the ``X-``? Can this just be ``IBM-1047-S390``?

Reply:
Should not have "-s390" suffix. Name of this charset should just be "IBM-1047"

Reply:
To clarify, (maybe add this as a code comment for future developers). Charset names should be taken from: https://www.iana.org/assignments/character-sets/character-sets.xhtml